### PR TITLE
[Bug]Arrange ColumnGroups in schema order

### PR DESCRIPTION
**Pull request by ashokblend**
1.While defining Column groups, if group is not defined in schema order than data loading is failing
   fix is done to arrange the group in schema order
2.Changed Sortindexwriter and SortIndexReader to get pathservice from factory and also changed visibility of cleanupOldSortIndex method to protected
3.Changed DictionaryReader to get pathService from factory and then get carbontable

### CODE REVIEW

**Comment by ravipesala** [comment]
Build finished.

File: integration/spark/src/main/scala/org/carbondata/spark/util/CommonUtil.scala [hunk 1]
```diff
@@ -167,4 +167,34 @@ object CommonUtil {
...
+   * @param colGrps
+   * @param dims
+   * @return columns of column groups in schema order
+   */
+  def arrangeColGrpsInSchemaOrder(colGrps: Seq[String], dims: Seq[Field]): Seq[String] = {
+    val sortedColGroups: Seq[String] = colGrps.sortBy {
+      colGrp =>
+        val firstCol = colGrp.split(",")(0)
+        val dimIndex: Int = getDimIndex(firstCol, dims)
+        dimIndex.toString()
```
Comment (gvramana) [inline integration/spark/src/main/scala/org/carbondata/spark/util/CommonUtil.scala]: comparision cannot be by string, it should be integer

**Comment by ashokblend** [comment]
@gvramana review comment is handled, please check.

**Comment by ravipesala** [comment]
Build finished.

**Comment by gvramana** [comment]
LGTM

**Pull request merged**